pep/app/services/pep-session.ts

```diff
@@ -16,7 +21,53 @@ export interface AuthenticatedData {
 export default class PepSessionService extends SessionService.extend({
     data: (oneWay('session.content') as unknown) as AuthenticatedData
 }) {
-    // normal class body definition here
+    @service cookies!: CookiesService;
+
+    get isCookieSecure() {
+        return Number(ENV.cookieSecure) === 1;
+    }
+
+    /**
+     * Set the unauthed session date in a cookie
+     *
+     * @param {PepSecureAuthenticatedData} sessionData
+     * @memberof PepSessionService
+     */
+    setUnauthenticatedSession(sessionData: PepSecureAuthenticatedData) {
+        const resultString = JSON.stringify(sessionData);
+        this.cookies.write(UNAUTHENTICATED_SESSION_COOKIE_NAME, resultString, {
+            domain: ENV.cookieDomain,
+            secure: this.isCookieSecure,
+            sameSite: ENV.cookieSameSite,
+            expires: new Date('2525-01-01') // never!!!
+        });
+    }
+
+    /**
+     * Get the unauthed session data
+     *
+     * @returns {(PepSecureAuthenticatedData | undefined)}
+     * @memberof PepSessionService
+     */
+    getUnauthenticatedSession(): PepSecureAuthenticatedData | undefined {
+        const cookie = this.cookies.read(UNAUTHENTICATED_SESSION_COOKIE_NAME, {
+            secure: this.isCookieSecure,
+            sameSite: ENV.cookieSameSite
+        });
+        return cookie ? JSON.parse(cookie) : undefined;
+    }
+
+    /**
+     * Clear the unauthed session data. This only happens when when the user successfully logs in
+     *
+     * @memberof PepSessionService
+     */
+    clearUnauthenticatedSession() {
```

Comment:
i may have missed its usage elsewhere, but we should be using this to clear the unauthed session data cookie when either authenticators completes a successful "real" authentication.